### ISSUE TITLE
User with @ in the name

### ISSUE THREAD

**Issue by ddavtian** [flyway/flyway#1775]
###### What version of Flyway are you using?

4.2.0

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin, SBT plugin, ANT tasks)

Command-line

###### What database are you using (type & version)?

mySQL 5.7

###### What operating system are you using?

OS X

My username is `octo@mike-xxx-db`if I use it as `-user=octo@mike-xxx-db` I see the ending being truncated so only `octo` is being sent. If I use it as `'-user=octo@mike-xxx-db'` I am getting `ERROR: Invalid operation: '-user=octo@mike-xxx-db'`

```
 $ ./flyway '-user=octo@mike-xxx-db' '-password=162xxxxxxxxxx' -url=jdbc:mysql://mike-xxx-db.mysql.database.azure.com:3306/octo migrate 1 ↵
Flyway 4.2.0 by Boxfuse

ERROR:
Unable to obtain Jdbc connection from DataSource (jdbc:mysql://mike-xxx-db.mysql.database.azure.com:3306/octo) for user 'octo@mike-xxx-db': Could not connect: Access denied for user 'octo'@'xx.120.xx.107' (using password: YES)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 28000
Error Code : 1045
Message    : Could not connect: Access denied for user 'octo'@'xx.120.xx.107' (using password: YES)
```
In this case Azure is expecting the entire user name for mySQL to be `octo@mike-xxx-db`

**Comment by axelfontaine** [comment]
Flyway passes the entire user name as is. This is either an issue with your driver or the database.

**Comment by ddavtian** [comment]
@axelfontaine 

The flyway command line tool use MariaDB Connector/J version 1.4.5. This version has a bug which is fixed at version 1.5.6 (change log https://mariadb.com/kb/en/library/mariadb-connector-j-156-release-notes/ ), bug ID CONJ-394 (https://jira.mariadb.org/browse/CONJ-394 ). This bug blocked the authentication phase between flyway and its server.
 
To mitigate the issue, download MariaDB Connector/J version 1.6.4, the recent updated driver version (https://mariadb.com/kb/en/library/about-mariadb-connector-j).
 
Replace the old connector with this one: `mariadb-java-client-1.6.4.jar`
